tools/kleaver/main.cpp

```diff
@@ -95,6 +95,11 @@ namespace {
   llvm::cl::opt<std::string> directoryToWriteQueryLogs("query-log-dir",llvm::cl::desc("The folder to write query logs to. Defaults is current working directory."),
 		                                               llvm::cl::init("."));
 
+  llvm::cl::opt<bool>
+  QueryIndependence("assume-query-independence",
```

Comment:
The current name has overloaded meanings which to me is confusing which I think should be avoided. Given that the default behaviour is reasonably sensible, I think the option to change the default behaviour should accurately reflect what is actually performed. How about...

``--clear-array-decls-after-query``

?